htdocs/compta/paiement/list.php

```diff
@@ -187,7 +187,7 @@
 } else {
 	// DISTINCT is to avoid duplicate when there is a link to sales representatives
 	$sql = "SELECT DISTINCT p.rowid, p.ref, p.datep, p.fk_bank, p.amount, p.statut, p.num_paiement";
-	$sql .= ", c.code as paiement_code";
```

Comment:
I do not experience any trouble on v18 branch.

Reply:
@eldy click here to order by type of payment 

<img width="811" alt="Capture d’écran 2024-01-19 à 22 27 35" src="https://github.com/Dolibarr/dolibarr/assets/454387/c3f5c63e-610a-4eed-8682-8d203407a355">

Reply:
@eldy i have this error when i click to reorder by type of payment, with a Mysql 5.7.39 not MariaDB !!

```
Requête dernier accès en base en erreur: SELECT DISTINCT p.rowid, p.ref, p.datep, p.fk_bank, p.amount, p.statut, p.num_paiement, c.code as paiement_code, ba.rowid as bid, ba.ref as bref, ba.label as blabel, ba.number, ba.account_number as account_number, ba.fk_accountancy_journal as accountancy_journal, s.rowid as socid, s.nom as name, s.email FROM llx_paiement as p LEFT JOIN llx_c_paiement as c ON p.fk_paiement = c.id LEFT JOIN llx_bank as b ON p.fk_bank = b.rowid LEFT JOIN llx_bank_account as ba ON b.fk_account = ba.rowid LEFT JOIN llx_paiement_facture as pf ON p.rowid = pf.fk_paiement LEFT JOIN llx_facture as f ON pf.fk_facture = f.rowid LEFT JOIN llx_societe as s ON f.fk_soc = s.rowid WHERE p.entity IN (1) ORDER BY c.libelle DESC LIMIT 21
Code retour dernier accès en base en erreur: DB_ERROR_3065
Information sur le dernier accès en base en erreur: Expression #1 of ORDER BY clause is not in SELECT list, references column 'dolibarr18.c.libelle' which is not in SELECT list; this is incompatible with DISTINCT
```

Reply:
@eldy my config of Mysql : 

for innodb_strict_mode	`ON`

for sql_mode : 
`ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION`